silk/tests/openthread/ot_test_insecure_traffic_join.py

```diff
@@ -0,0 +1,158 @@
+                                           channel=random.randint(11, 25),
+                                           fabric_id="{0:06x}dead".format(random.randint(0, 0xffffff)))
+
+        cls.thread_sniffer_init(cls.network_data.channel)
+
+    @classmethod
+    @testcase.teardown_class_decorator
+    def tearDownClass(cls: 'TestInsecureTrafficJoin'):
+        for device in cls.device_list:
+            device.tear_down()
+
+    @testcase.setup_decorator
+    def setUp(self):
+        pass
+
+    @testcase.teardown_decorator
+    def tearDown(self):
+        pass
+
+    def transmit_receive_udp_message(self, src, dst, src_address, dst_address, port, src_port):
+        timeout = 5
+        delay = 1
+        message = random_string(10)
+
+        if src_address == src.ip6_lla:
+            src_address = f"{src_address}%{src.netns}"
+
+        dst.receive_udp_data(port, message, timeout)
+        time.sleep(delay)
+        src.send_udp_data(target=dst_address, port=port, message=message, source=src_address, src_port=src_port)
+        time.sleep(timeout - delay)
+
+    @testcase.test_method_decorator
+    def test01_pairing(self):
+        # allowlisting between parent (i.e. leader) and router
+        self.parent.allowlist_node(self.router)
+        self.router.allowlist_node(self.parent)
+
+        self.parent.form(self.network_data, "router")
+        self.parent.permit_join_new(duration_sec='100', port=INSECURE_PORT)
+        self.wait_for_completion(self.device_list)
+
+        self.assertEqual(self.parent.get(wpan.WPAN_NETWORK_ALLOW_JOIN), 'true')
+
+        self.logger.info(self.parent.ip6_lla)
+        self.logger.info(self.parent.ip6_thread_ula)
+
+        self.network_data.xpanid = self.parent.xpanid
+        self.network_data.panid = self.parent.panid
+
+        # Join parent network from router without setting the key
+        self.router.join_node(self.network_data, "router", should_set_key=False)
+        self.wait_for_completion(self.device_list)
+
+    @testcase.test_method_decorator
+    def test02_verify_router_joins_parent_network(self):
+        self.assertEqual(self.router.getprop(wpan.WPAN_STATE), wpan.STATE_CREDENTIALS_NEEDED)
+        self.assertEqual(self.router.getprop(wpan.WPAN_NAME), self.parent.getprop(wpan.WPAN_NAME))
+        self.assertEqual(self.router.getprop(wpan.WPAN_XPANID), self.parent.getprop(wpan.WPAN_XPANID))
+        self.assertEqual(self.router.getprop(wpan.WPAN_PANID), self.parent.getprop(wpan.WPAN_PANID))
+
+    @testcase.test_method_decorator
+    def test03_transmit_receive_insecure_traffic(self):
+        # Send insecure traffic from router to parent using link-local IP address
+        # for src/dst and insecure port number
+        self.transmit_receive_udp_message(self.router, self.parent, self.router.ip6_lla, self.parent.ip6_lla,
+                                          src_port=SRC_PORT, port=INSECURE_PORT)
+
```

Comment:
Please check the SRC_PORT obtained here.

From toranj: for insecure traffic we need to find the receiver's random src port too.
i.e. "Get the random src port number used by node1" (https://github.com/openthread/openthread/blob/master/tests/toranj/test-009-insecure-traffic-join.py#L88) and then in (https://github.com/openthread/openthread/blob/master/tests/toranj/test-009-insecure-traffic-join.py#L93) the rx_port is associated with node2. This made rx_port a little confusing. 

Please find below the code snippet being discussed. Thanks


"Send insecure traffic from node2 to node1 using link-local IP address
for src/dst and insecure port number"

sender = node2.prepare_tx(ll2, (ll1, insecure_port), "Hi (insecure)", NUM_MSGS)
recver = node1.prepare_rx(sender)
wpan.Node.perform_async_tx_rx()
verify(sender.was_successful)
verify(recver.was_successful)


"Get the random src port number used by node1 and ensure node2 allows
insecure rx traffic on that port"

rx_port = recver.all_rx_msg[0][1][1]
node2.permit_join(duration_sec='100', port=str(rx_port))



"Send insecure reply from node1 to node2"

sender2 = node1.prepare_tx((ll1, insecure_port), (ll2, rx_port), "Hi back! (insecure)", NUM_MSGS)
recver2 = node2.prepare_rx(sender2)
wpan.Node.perform_async_tx_rx()
verify(sender2.was_successful)
verify(recver2.was_successful)

Reply:
@abtink, can you help take a look?

Reply:
The flow of the test as follows:
- We have two nodes. 
- Node 1 has a network (attached) and node 2 attaches insecurely (not knowing the master key)
- On node 1 we allow specific `insecure_port` number (1234)
   - This means node1 is allowed to receive message where dst port is `insecure_port` (1234) (with any src port).
- We then send a message from node 2 to node 1, where we use `insecure_port` as dst port and a random port number as src port.
   - We expect and check that node 1 does indeed receive this ok.
- Now on node 1 we check what the src port of the received msg is (which is saved as `rx_port`).
- Then on node 2, we allow the `rx_port` (`node2.permit_join(duration_sec='100', port=str(rx_port))`
   - This again means, node 2 now can receive/accept insecure traffic where dst port is set to `rx_port` value
- Finally we send an insecure msg this time from node 1 to node 2 using `rx_port` as dst port.

What we care is to use two different port number (for node 1 and node 2). We can also consider using two different hard-coded port values (no need to use random one)

Reply:
Thanks for the detailed explanation Abtin. This clarifies the doubt about ports. And I think the test logic used to define SRC_PORT in advance is doing the same thing.